variables.tf

```diff
@@ -1,120 +1,393 @@
+  description = "Maximum number of NAT Gateways or NAT instances to create"
+  default     = 999
 }
 
-variable "igw_id" {
+variable "private_subnets_enabled" {
+  type        = bool
+  description = "If false, do not create private subnets (or NAT gateways or instances)"
+  default     = true
+}
+
+variable "public_subnets_enabled" {
+  type        = bool
+  description = <<-EOT
+    If false, do not create public subnets.
+    Since NAT gateways and instances must be created in public subnets, these will also not be created when `false`.
+    Since the only default route for a private subnet is to a NAT gateway or instance,
+    route tables and routes will also not be created when `false`.
+    EOT
+  default     = true
+}
+
+variable "private_label" {
   type        = string
-  description = "Internet Gateway ID the public route table will point to (e.g. `igw-9c26a123`)"
+  description = "The string to use in IDs and elsewhere to distinguish resources for the private subnets from resources for the public subnets"
+  default     = "private"
 }
 
-variable "cidr_block" {
+
+variable "public_label" {
   type        = string
-  description = "Base CIDR block which will be divided into subnet CIDR blocks (e.g. `10.0.0.0/16`)"
+  description = "The string to use in IDs and elsewhere to distinguish resources for the private subnets from resources for the public subnets"
+  default     = "public"
 }
 
-variable "availability_zones" {
+variable "ipv4_enabled" {
+  type        = bool
+  description = "Set true to enable IPv4 addresses in the subnets"
+  default     = true
+}
+
+variable "ipv6_enabled" {
+  type        = bool
+  description = "Set true to enable IPv6 addresses in the subnets"
+  default     = false
+}
+
+variable "ipv4_cidr_block" {
   type        = list(string)
-  description = "List of Availability Zones where subnets will be created"
+  description = <<-EOT
+    Base IPv4 CIDR block which will be divided into subnet CIDR blocks (e.g. `10.0.0.0/16`). Ignored if `ipv4_cidrs` is set.
+    If no CIDR block is provided, the VPC's default IPv4 CIDR block will be used.
+    EOT
+  default     = []
+  validation {
+    condition     = length(var.ipv4_cidr_block) < 2
+    error_message = "Only 1 ipv4_cidr_block can be provided. Use ipv4_cidrs to provide a CIDR per subnet."
+  }
+}
 
+variable "ipv6_cidr_block" {
+  type        = list(string)
+  description = <<-EOT
+    Base IPv6 CIDR block from which `/64` subnet CIDRs will be assigned. Must be `/56`. (e.g. `2600:1f16:c52:ab00::/56`).
+    Ignored if `ipv6_cidrs` is set. If no CIDR block is provided, the VPC's default IPv6 CIDR block will be used.
+    EOT
+  default     = []
   validation {
-    condition     = length(var.availability_zones) > 0
-    error_message = "Availability zones must be greater than zero."
+    condition     = length(var.ipv6_cidr_block) < 2
+    error_message = "Only 1 ipv6_cidr_block can be provided. Use ipv6_cidrs to provide a CIDR per subnet."
```

Comment:
nit: consider `only one` instead of the numeral `1`

Reply:
Why? In technical writing, quantities should be expressed in numerals for consistency. "1, 11, 111", not "one, 11, 111" or "one, eleven, one hundred eleven".